src/Summoner/CLI.hs

```diff
@@ -128,14 +169,31 @@ summonerVersion = toString $ intercalate "\n" $ [sVersion, sHash, sDate] ++ [sDi
 -- All possible commands.
 summonerP :: Parser Command
 summonerP = subparser
-    ( command "new"
-      (info (helper <*> newP) $ progDesc "Create a new Haskell project")
-    )
+    $ command "new" (info (helper <*> newP) $ progDesc "Create a new Haskell project")
+   <> command "show" (info (helper <*> showP) $ progDesc "Show available licenses or ghc versions")
 
 ----------------------------------------------------------------------------
 -- New command parsers
 ----------------------------------------------------------------------------
 
+-- | Parses options of the @show@ command.
+showP :: Parser Command
+showP = ShowInfo <$> (ghcP <|> ShowLicense <$> licenseP)
+
+ghcP :: Parser ShowOpts
+ghcP = flag' ShowGhc $ long "ghc" <> short 'g' <> help "Show available ghc versions"
+
+licenseP :: Parser LicenseList
+licenseP = subparser $
+    command "licenses" (info (helper <*> licenseList) $ progDesc "Show available licenses")
+
+licenseList :: Parser LicenseList
+licenseList =  LicenseList <$> optional licenseText
+
+licenseText :: Parser String
+licenseText = strOption $
+    long "name" <> short 'n' <> metavar "LICENSE_NAME" <> help "Show license text"
```

Comment:
I guess there should be a way to avoid this `name` option, so we can:
```
summon show --licence
# and
summon show --licence MIT
```

Reply:
I would like to have `stack`-like interface for this. For example, in `stack` there's `ls` command that shows different things. And I can do:

```shell
$ stack ls # prints error that command is missing
$ stack ls dependencies # shows dependencies
$ stack ls snapshots # shows snapshots
```

It would be great if we can write something like that. Specifically, I propose to have the following interface:

```shell
$ summon show # error
$ summon show ghc
$ summon show licenses
$ summon show license MIT
```
Also, what do you think about renaming `show` to `ls`?

Reply:
> $ summon show license MIT
> summon show --licence MIT

I am not sure is it possible to have `optional metavar`. Task is to have double functionality. 
First `summon show license` shows list of licenses. And second `summon show license MIT` shows particular license`s text.
And I am not sure is `command` like `summon show license MIT` has **metavar** possibility or not.

Reply:
@willbasky It's okay to have two different words for different commands:

* `licenses` to show list of all supported licences
* `license` to show only specific license

Also, I'm thinking about renaming `show ghc` to `show ghcs` to make naming more consistent.

Reply:
About `ls` or `show`. We need data what users prefers or some arguments why one is better then other one.

Reply:
`ls` is shorter, but usually `ls` is used to display multiple things (like entries in directory), so it might be confusing to have `summon ls license MIT`
`show` is more human readable, but longer. Though, only 2 characters longer.